secure_message/common/utilities.py

```diff
@@ -78,30 +79,44 @@ def process_paginated_list(paginated_list, host_url, user, message_args, endpoin
 
 
 def add_to_details(messages):
-    """Adds a @msg_to key every message in a list of messages.
+    """Adds a @msg_to key to every message in a list of messages.
     Every msg_to uuid is resolved to include details of the user.
 
     If the call for the internal user id fails, an exception will be thrown.
     If the external user id cannot be found in the list that we got from the party service.  There
     won't be a @msg_to value returned in the payload.  The API documentation notes that these elements
     aren't guaranteed to be provided so we're not breaking the contract by doing this.
+
+    Note:28th Feb 2019. Try except blocks added to try and identify cause of IndexError seen in splunk logs
+    No functionality added, just log and raise.
     """
     external_user_details = {}
-    for user in party.get_users_details(get_external_user_uuid_list(messages)):
+
+    try:
+        external_users = get_external_user_uuid_list(messages)
+    except IndexError:
+        logger.exception(f"Exception getting external user uuids for messages:'{json.dumps(messages)}'")
+        raise
+
+    for user in party.get_users_details(external_users):
         external_user_details[user['id']] = user
 
     for message in messages:
-        if not message["from_internal"]:
-            message.update({"@msg_to": [internal_user_service.get_user_details(message["msg_to"][0])]})
-        else:
-            if external_user_details.get(message['msg_to'][0]):
-                message.update({'@msg_to': [external_user_details.get(message['msg_to'][0])]})
+        try:
+            if not message["from_internal"]:
+                message.update({"@msg_to": [internal_user_service.get_user_details(message["msg_to"][0])]})
+            else:
+                if external_user_details.get(message['msg_to'][0]):
+                    message.update({'@msg_to': [external_user_details.get(message['msg_to'][0])]})
+        except IndexError:
+            logger.exception(f"Exception adding to details message:'{json.dumps(message)}'")
```

Comment:
Same as the other message.  Also, thinking about it, you don't want to dump out a giant blob of json in the logs as it's difficult to look at.  We should target the things we might care about and increase it (probably in a new field for easier searching) if it's not enough

Reply:
This is a fishing expedition which is why it is so broad , but will rethink in light of not logging personal data